src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -248,8 +248,9 @@ public void OnRenderCompleted(long incomingBatchId, string errorMessageOrNull)
 
                 if (lastBatchId < incomingBatchId)
                 {
-                    HandleException(
-                        new InvalidOperationException($"Received an acknowledgement for batch with id '{incomingBatchId}' when the last batch produced was '{lastBatchId}'."));
+                    // This exception is due to a bad client input, so we mark it as such to prevent loging it as a warning and
+                    // flooding the logs with warnings.
+                    throw new RemoteRendererException(badInput: true, $"Received an acknowledgement for batch with id '{incomingBatchId}' when the last batch produced was '{lastBatchId}'.");
```

Comment:
Why have a `bool` flag on `RemoteRendererException` instead of a more specific exception type like `InvalidAcknowledgementBatchIdException`?

Reply:
Because it's internal, cheaper than having another type and is not going to be used inside a catch block, so there's no value of having an additional exception.

Reply:
My opinion is that exception hierarchies are good when you have public consumers or when you use them to handle different scenarios in code through catch blocks, when that's not the case, having an additional type is less valuable.

Reply:
Removed the bool, switched to use specific exceptions

Reply:
Used a different exception for this.